pages/api/applicants.ts

```diff
@@ -0,0 +1,48 @@
+import type { Applicant } from './lib/applicant';
+import type { NextApiRequest, NextApiResponse } from 'next';
+import { getDB } from './lib/db';
+
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse<{ applicants: Applicant[], total: number } | { message: string }>
+) {
+  const db = await getDB();
+
+  const { name, sortColumn, sortOrder, filter, page = 1, pageSize = 10 } = req.body;
+  let query = "SELECT * FROM applicant WHERE 1=1";
+  let countQuery = "SELECT COUNT(*) as count FROM applicant WHERE 1=1";
```

Comment:
This came about when adding pagination; I needed a way to disable the "next" button if there would be no results on it, and having a total count allowed for this logic:

`disabled={page * pageSize >= totalApplicants}`

Which is definitely pretty basic. My first next step for a large dataset would likely be to at least have the count pre-calculated in its own column or table so that the whole table isn't scanned with every request. 

It would also make sense for the back end to handle the logic of whether the current response has any pages after it, and just send a boolean like `hasMore`.

Although, this all also depends on our pagination strategy as a whole. Depending on the size and shape of data, and how fast a page response can be returned, we might decide that we want to send everything to the front end at once. I've been part of accomplishing this before with a combination of redis and gzip, which also led to us to use virtualization on the front end to make sure that the DOM was only ever populated with what the user saw.